tls/s2n_handshake_io.c

```diff
@@ -622,7 +654,9 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
         if (!conn->config->quic_enabled) {
             conn->handshake.handshake_type |= MIDDLEBOX_COMPAT;
         }
-        conn->handshake.handshake_type |= FULL_HANDSHAKE;
+        if (conn->psk_params.chosen_psk == NULL) {
+            conn->handshake.handshake_type |= FULL_HANDSHAKE;
```

Comment:
You need to prevent a CLIENT_AUTH handshake without FULL_HANDSHAKE. You don't have handshakes for that, and it's not allowed. You'll need to consider the client side case too: https://github.com/awslabs/s2n/blob/main/tls/s2n_handshake_io.c#L1030-L1036